indra/newview/llfloaterimnearbychat.cpp

```diff
@@ -606,6 +608,13 @@ void LLFloaterIMNearbyChat::sendChat( EChatType type )
                 if(!LLGestureMgr::instance().triggerAndReviseString(utf8text, &utf8_revised_text))
                 {
                     utf8_revised_text = utf8text;
+                    // check if the message is /filename.lua and execute the Lua script
+                    static const boost::regex is_lua_script("^/.*\\.luau?$");
```

Comment:
I recommend using `ll_regex_match()` from llregex.h: `boost::regex_match()` can throw. `ll_regex_match()` defends against that case.

Also, now that `runScriptFile()` can pass command-line arguments to the subject Lua script, I would match even if there's more text beyond the `.lua` extension.

Reply:
Thanks. You're right about passing arguments.